### PR TITLE
UpdateSpec properties should be optional

### PR DESCRIPTION
**Pull request by tugend**
#4268

### CODE REVIEW

File: react/react-addons-global.d.ts [hunk 1]
```diff
@@ -77,9 +77,9 @@ declare module React {
     // ----------------------------------------------------------------------
 
     interface UpdateSpec {
-        $set: any;
-        $merge: {};
-        $apply(value: any): any;
+        $set?: any;
+        $merge?: {};
+        $apply(value: any)?: any;
```
Comment (vvakame) [inline react/react-addons-global.d.ts]: `$apply(value: any)?: any;` to `$apply?(value: any): any;`

File: react/react-addons.d.ts [hunk 2]
```diff
@@ -812,9 +812,9 @@ declare module "react/addons" {
     // ----------------------------------------------------------------------
 
     interface UpdateSpec {
-        $set: any;
-        $merge: {};
-        $apply(value: any): any;
+        $set?: any;
+        $merge?: {};
+        $apply(value: any)?: any;
```
Comment (vvakame) [inline react/react-addons.d.ts]: `$apply(value: any)?: any;` to `$apply?(value: any): any;`

**Comment by vvakame** [comment]
@tugend thank you for your contributions.
please travis ci failed.

**Comment by tugend** [comment]
@vvakame, whoops, sorry about that.
I will be more carefull in the future! ^_^

**Comment by vvakame** [comment]
@tugend thanks mate!

**Pull request merged**